### PR TITLE
MOB-4657 add card save switch

### PR DESCRIPTION
**Pull request by BartlomiejSopalaKarhoo**
## JIRA
[MOB-4657](https://karhoo.atlassian.net/browse/MOB-4657)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

[MOB-4657]: https://karhoo.atlassian.net/browse/MOB-4657?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BraintreePSP/BraintreePaymentScreensBuilder.swift [hunk 1]
```diff
@@ -25,12 +25,18 @@ final public class BraintreePaymentScreenBuilder: PaymentScreenBuilder {
     public init(){}
 
     public func buildAddCardScreen(
+        guestMode: Bool,
```
Comment (AleksanderWedrychowskiKarhoo) [inline BraintreePSP/BraintreePaymentScreensBuilder.swift:28]: I would change the name of this parameter, to make it clear this value affects a payment method's saving behavior.

**Comment by BartlomiejSopalaKarhoo** [comment]
A added cardholder name. It is present on Android too. It is not clear yet how backend should handle it, but we need to enable this String.

File: BraintreePSP/BraintreePaymentScreensBuilder.swift [hunk 2]
```diff
@@ -25,12 +25,19 @@ final public class BraintreePaymentScreenBuilder: PaymentScreenBuilder {
...
+        allowToSaveCard: Bool,
         paymentsToken: PaymentSDKToken,
         paymentMethodAdded: ScreenResultCallback<Nonce>?,
         flowItemCallback: ScreenResultCallback<Screen>?
     ) {
         let request = BTDropInRequest()
         request.uiCustomization = getUiCustomization()
+        request.cardholderNameSetting = .required
+        if allowToSaveCard {
+            request.allowVaultCardOverride = true
```
Comment (dianakarh) [inline BraintreePSP/BraintreePaymentScreensBuilder.swift:37]: This looks a bit weird. Shouldn't request.vaultCard be set in both cases? Like this:
```
if allowToSaveCard {
            request.vaultCard = true
            request.allowVaultCardOverride = true
} else {
            request.vaultCard = false
        }
```

Comment (BartlomiejSopalaKarhoo) [reply BraintreePSP/BraintreePaymentScreensBuilder.swift:37]: According to docs (https://braintree.github.io/braintree-ios-drop-in/current/Classes/BTDropInRequest.html) all variables have default value:
vaultCard is true by default
and 
allowVaultCardOverride is false. 
So we override this value only when required.

**Review by dianakarh — APPROVED** [review]

**Review by AleksanderWedrychowskiKarhoo — APPROVED** [review]

**Pull request merged**